backend/inspirehep/accounts/views.py

```diff
@@ -104,3 +109,21 @@ def set_orcid_push_setting():
         )
 
     return jsonify({"message": "Successfully changed orcid push setting"}), 200
+
+
+@blueprint.route("/settings/update-email", methods=["POST"])
+@login_required
+@parser.use_args(
+    {
+        "new_email": fields.String(required=True),
+    },
+    locations=("json",),
+)
+def set_email(args):
+    new_email = args["new_email"]
+    try:
+        current_user.email = new_email
```

Comment:
I'm not sure this is enough, it's a proxy object

Reply:
I was not sure either, found this source where they set `current_user` params like that https://stackoverflow.com/questions/62361924/flask-sqlalchemy-update-the-current-user (there's also a test in the pr). But to be on the safe side I'll do it with db table query.